src/hist/namedhist.py

```diff
@@ -36,38 +35,24 @@ def fill(self, *args, **kwargs):
             NamedHist object. NamedHist could only be filled by names.
         """
 
-        indices = []
-        values = []
-        for name, val in kwargs.items():
-            for index, axis in enumerate(self.axes):
-                if name == axis.name:
-                    indices.append(index)
-                    values.append(val)
-                    break
-            else:
-                raise ValueError("The axis names could not be found")
+        if len(kwargs) and all(isinstance(k, str) for k in kwargs.keys()):
+            return super().fill(*args, **kwargs)
 
-        d = dict(zip(indices, values))
-        lst = sorted(d.items(), key=lambda item: item[0])
-        nd = np.asarray(lst, dtype=object)
-        data = nd.ravel()[1::2]
-        super().fill(*data)
-
-        return self
+        else:
+            raise TypeError(
+                f"Only fill by names are supported for {self.__class__.__name__}"
+            )
 
     def __getitem__(self, index):
         """
             Get histogram item.
         """
 
         if isinstance(index, dict):
-            k = list(index.keys())
-            if isinstance(k[0], str):
-                for key in k:
-                    for idx, axis in enumerate(self.axes):
-                        if key == axis.name:
-                            index[idx] = index.pop(key)
-                            break
+            if all(isinstance(k, int) for k in index.keys()):
```

Comment:
You mean `any`, not `all`, right? Or you can use `not all(isinstance(k, str)...`

Reply:
Yep! Should be any.